pkg/reconciler/votereconciler/redis_create.go

```diff
@@ -13,6 +13,15 @@ func (r *Redis) CreateVote(inp []*votestorage.Object) ([]*votestorage.Object, er
 	var err error
 
 	for i := range inp {
+		{
+			if inp[i].Hash == "" {
+				inp[i].Lifecycle = "pending"
+			}
+			if inp[i].Hash != "" {
+				inp[i].Lifecycle = "onchain"
+			}
+		}
```

Comment:
Here we infer the resource lifecycle for votes based on the provided transaction hash.